lib/adagrams.rb

```diff
@@ -0,0 +1,112 @@
+require "csv"
+
+def draw_letters
+  letter_pool = []
+  number_of_letters = [9, 2, 2, 3, 12, 2, 3, 2, 9, 1, 1, 4, 2, 6, 8, 2, 1, 6, 4, 6, 4, 2, 2, 1, 2, 1]
+  letters = ("A".."Z").to_a
+  i = 0
+  while i < 26
+    number_of_letters[i].times do
+      letter_pool.push(letters[i])
+    end
+    i = i + 1
+  end
+
+  return letter_pool.sample(10)
+end
+
+puts "Using the following letters, create an Adagram"
+hand = draw_letters
+puts hand
+
+puts "What is your Adagram?:"
+adagram_input = gets.chomp.to_s.upcase
+
+def uses_available_letters?(input, letters_in_hand)
+  letters_array = input.chars
```

Comment:
There doesn't seem to me a good reason to make this re-assignment. In your while you could have just as easily done `input.chars.length`.